src/main/scala/strawman/collection/immutable/LazyList.scala

```diff
@@ -36,6 +35,16 @@ class LazyList[+A](expr: => LazyList.Evaluated[A])
 
   protected[this] def fromSpecificIterable(coll: collection.Iterable[A]): LazyList[A] = fromIterable(coll)
 
+  protected[this] def toCollection: LazyList[A] => IterableOnce[A] = identity
+
+  def updated[B >: A](index: Int, elem: B): LazyList[B] =
+    zipWithIndex.map { case (a, i) => if (i == index) elem else a }
+
+  def zipWithIndex: LazyList[(A, Int)] =
```

Comment:
`unfold` does some wrapping that could be avoided. We can probably also get rid of the `Option` in the `LazyList` encoding and use null instead.